src/main/java/net/snowflake/client/core/HttpUtil.java

```diff
@@ -69,16 +69,25 @@
 
   public static final String JDBC_TTL = "net.snowflake.jdbc.ttl";
 
-  /** The unique httpClient shared by all connections. This will benefit long- lived clients */
-  private static Map<OCSPMode, CloseableHttpClient> httpClient = new ConcurrentHashMap<>();
+  /**
+   * The unique httpClient shared by all connections. This will benefit long- lived clients. Key =
+   * proxy host + proxy port + nonProxyHosts, Value = Map of [OCSPMode, HttpClient]
+   */
+  public static Map<HttpClientSettingsKey, CloseableHttpClient> httpClient =
+      new ConcurrentHashMap<>();
 
   /**
-   * The unique httpClient shared by all connections that don't want decompression. This will
-   * benefit long-lived clients
+   * The unique httpClient map shared by all connections that don't want decompression. This will
+   * benefit long-lived clients. Key = proxy host + proxy port + nonProxyHosts, Value = Map
+   * [OCSPMode, HttpClient]
    */
-  private static Map<OCSPMode, CloseableHttpClient> httpClientWithoutDecompression =
+  private static Map<HttpClientSettingsKey, CloseableHttpClient> httpClientWithoutDecompression =
       new ConcurrentHashMap<>();
 
+  /** The map of snowflake route planners */
+  private static Map<HttpClientSettingsKey, SnowflakeMutableProxyRoutePlanner>
```

Comment:
The RoutePlanner is used as part of the HttpClient construction. It may have to change later if the nonProxyHosts value changes, so it's stored as part of a hashmap and updated whenever nonProxyHosts changes.